src/hooks/useMixpanel.ts

```diff
@@ -0,0 +1,21 @@
+// eslint-disable-next-line no-restricted-imports
+import mixpanel from "mixpanel-browser";
+import { useMemo } from "react";
+
+// https://stackoverflow.com/questions/32203420/check-if-mixpanel-library-has-been-loaded
+const isLoaded = () => mixpanel.hasOwnProperty("get_distinct_id");
+
+const noopTrack: typeof mixpanel.track = () => {};
+
+/**
+ * Returns an object to allow us to expose more of Mixpanel's API in future.
+ *
+ * NOTE: Extra level of function indirection ensures `isLoaded` is evaluated not just the first time.
+ */
+export const useMixpanel = () =>
+  useMemo(
+    () => ({
+      track: () => (isLoaded() ? mixpanel.track.bind(mixpanel) : noopTrack),
```

Comment:
I think as is this may not work correctly.. as I forgot that we probably want to pass parameters through to mixpanel's track function.. so we need to add whatever params it accepts here as well.